homeassistant/components/modbus/cover.py

```diff
@@ -80,7 +79,9 @@ def __init__(
         self._state_opening = config[CONF_STATE_OPENING]
         self._status_register = config.get(CONF_STATUS_REGISTER)
         self._status_register_type = config[CONF_STATUS_REGISTER_TYPE]
-        self._scan_interval = timedelta(seconds=config[CONF_SCAN_INTERVAL])
+        self._scan_interval = hub.control_scan_delay(
```

Comment:
Maybe put the scan interval attribute on the hub instead?

Reply:
There are only 1 hub for multiple platforms that each can have their own scan_interval, so that would mean I had to keep a list in the hub.

It is important to be able to configure the scan_interval at platform level, example:
1 sensor that monitors power usage in the house, with a scan_interval of a few seconds (to e.g. control extra heaters)
1 sensor that makes a graph of solar production, with a scan_interval of 15minutes.
1 switch with a sensor, here the scan_interval should really be -1 (future PR), but make it 1hour.

Please remember that modbus can normally have many platforms defined, some run in the hundreds, for such a configuration you really need to play with the scan_interval.